trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesTraitsService.php

```diff
@@ -229,20 +257,21 @@ public function insertTrait(array $trait, ?string $schema = NULL) {
       else {
         // A new trait asset, create a record.
         $rec = [
-          'id' => $genus_config['database']['name'] . ':' . $values['name'],
-          'name' => $values['name'],
-          'cv_name' => $genus_config[$type]['name'],
-          'definition' => $values['description'],
+          'db.name' => $genus_config['database']['name'],
+          'cv.name' => $genus_config[$type]['name'],
+          'dbxref.accession' => $values['name'],
+          'cvterm.name' => $values['name'],
+          'cvterm.definition' => $values['description'],
         ];
 
-        $ins = chado_insert_cvterm($rec, [], $schema);
+        $ins = $this->cvterm_buddy->upsertCvterm($rec, []);
         if (!$ins) {
           // Could not insert cvterm.
           throw new \Exception('A database error occurred while inserting a term.
-          Failed to insert term ' . $type . ' : ' . $term);
+          Failed to insert term ' . $type . ' : ' . $rec);
```

Comment:
This won't print as expected since `$rec` is an array. If $type = MyType, then you would see printed:
```
A database error occurred while inserting a term. Failed to insert term MyType : Array
```
I think what you want here is the cvterm name instead of the whole record, in which case, I would replace `$rec` with `$rec['cvterm.name']`.